File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/dataclasses/dataClassWithReservedWord/expected/types/Person.java
package com.netflix.graphql.dgs.codegen.java.testcases.dataclasses.dataClassWithReservedWord.expected.types;

import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Objects;

public class Person {
  private String info;

  private String _interface;

  public Person() {
  }

  public Person(String info, String _interface) {
    this.info = info;
    this._interface = _interface;
  }

  public String getInfo() {
    return info;
  }

  public void setInfo(String info) {
    this.info = info;
  }

  public String getInterface() {
    return _interface;
  }

  public void setInterface(String _interface) {
    this._interface = _interface;
  }

  @Override
  public String toString() {
    return "Person{info='" + info + "', interface='" + _interface + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    Person that = (Person) o;
    return Objects.equals(info, that.info) &&
        Objects.equals(_interface, that._interface);
  }

  @Override
  public int hashCode() {
    return Objects.hash(info, _interface);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private String info;

    private String _interface;

    public Person build() {
      Person result = new Person();
      result.info = this.info;
      result._interface = this._interface;
      return result;
    }

    public Builder info(String info) {
      this.info = info;
      return this;
    }

    public Builder _interface(String _interface) {
      this._interface = _interface;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/dataClassWithReservedWord/expected/types/Person.kt
package com.netflix.graphql.dgs.codegen.cases.dataClassWithReservedWord.expected.types

import com.fasterxml.jackson.`annotation`.JsonIgnoreProperties
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.fasterxml.jackson.`annotation`.JsonTypeInfo
import com.fasterxml.jackson.databind.`annotation`.JsonDeserialize
import com.fasterxml.jackson.databind.`annotation`.JsonPOJOBuilder
import com.netflix.graphql.dgs.codegen.cases.dataClassWithReservedWord.expected.Generated
import java.lang.IllegalStateException
import java.util.Objects
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String
import kotlin.jvm.JvmName

@Generated
@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
@JsonDeserialize(builder = Person.Builder::class)
public class Person(
  info: () -> String? = infoDefault,
  `interface`: () -> String? = interfaceDefault,
) {
  private val __info: () -> String? = info

  private val __interface: () -> String? = `interface`

  @get:JvmName("getInfo")
  public val info: String?
    get() = __info.invoke()

  @get:JvmName("getInterface")
  public val `interface`: String?
    get() = __interface.invoke()

  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (other !is Person) return false
    return (__info === infoDefault) == (other.__info === infoDefault) && (__info === infoDefault ||
        Objects.equals(info, other.info)) &&
    (__interface === interfaceDefault) == (other.__interface === interfaceDefault) && (__interface
        === interfaceDefault || Objects.equals(`interface`, other.`interface`))
  }

  override fun hashCode(): Int = Objects.hash(if (__info === infoDefault) infoDefault else info,
  if (__interface === interfaceDefault) interfaceDefault else `interface`)

  @Generated
  public companion object {
    private val infoDefault: () -> String? = 
        { throw IllegalStateException("Field `info` was not requested") }

    private val interfaceDefault: () -> String? = 
        { throw IllegalStateException("Field `interface` was not requested") }
  }

  @Generated
  @JsonPOJOBuilder
  @JsonIgnoreProperties("__typename")
  public class Builder {
    private var info: () -> String? = infoDefault

    private var `interface`: () -> String? = interfaceDefault

    @JsonProperty("info")
    public fun withInfo(info: String?): Builder = this.apply {
      this.info = { info }
    }

    @JsonProperty("interface")
    public fun withInterface(`interface`: String?): Builder = this.apply {
      this.`interface` = { `interface` }
    }

    public fun build(): Person = Person(
      info = info,
      `interface` = `interface`,
    )
  }
}
